source/slang/slang-check-decl.cpp

```diff
@@ -9308,8 +9308,161 @@ namespace Slang
             this, funcDecl, attr, DeclAssociationKind::PrimalSubstituteFunc);
     }
 
+    static int32_t findLevelsOfInheritance(AggTypeDecl* decl, int32_t inheritanceLevel, List<AggTypeDecl*>& linearInheritanceHierarchy)
+    {
+        // We need to assume multiple `InheritanceDecl` are possible per `decl` member list.
+        int32_t inheritanceDepth = inheritanceLevel;
+        for (auto maybeInheritance : decl->members)
+        {
+            auto inheritance = as<InheritanceDecl>(maybeInheritance);
+            if (!inheritance)
+                continue;
+            auto isValidInheritance = as<DeclRefType>(inheritance->base.type);
+            if (!isValidInheritance || !isValidInheritance->getDeclRef())
+                continue;
+            auto isInherited = as<AggTypeDecl>(isValidInheritance->getDeclRef().getDecl());
+            if (!isInherited)
+                continue;
+            linearInheritanceHierarchy.add(isInherited);
+            int32_t depthFound = findLevelsOfInheritance(isInherited, inheritanceLevel-1, linearInheritanceHierarchy);
+            if (depthFound < inheritanceDepth)
+                inheritanceDepth = depthFound;
+        }
+        return inheritanceDepth;
+    }
+
     void SemanticsDeclAttributesVisitor::visitStructDecl(StructDecl* structDecl)
     {
+        List<AggTypeDecl*> linearInheritanceHierarchy;
+        linearInheritanceHierarchy.add(structDecl);
+        int32_t levelsOfInheritance = findLevelsOfInheritance(structDecl, 0, linearInheritanceHierarchy);
+
+        List<ConstructorDecl*> ctorList;
+        bool fillCtorList = false;
+
+        auto findAllCtor = [&]() {
+            ConstructorDecl* ctorDefault = nullptr;
+            fillCtorList = true;
+            for (auto maybeCtor : structDecl->members)
+            {
+                auto ctor = as<ConstructorDecl>(maybeCtor);
+                if (!ctor)
```

Comment:
we seems prefer to use like this
```
if (auto ctor = as<ConstructorDecl>(maybeCtor))
{
               ...
}
```

Reply:
The following code structure was done to avoid having 2-3 extra indentations in-favor for an early leave